src/test/java/org/opensrp/web/rest/StockResourceTest.java

```diff
@@ -190,11 +214,153 @@ public void testFilter() {
+		assertEquals(result.getResponse().getContentAsString(), "");
+	}
+
+	@Test
+	public void testDelete() throws Exception {
+		ArgumentCaptor<Long> argumentCaptor = ArgumentCaptor.forClass(Long.class);
+		mockMvc.perform(MockMvcRequestBuilders.delete(BASE_URL + "/{id}", 1))
+				.andExpect(MockMvcResultMatchers.status().isNoContent()).andReturn();
+		Mockito.verify(stockService, Mockito.times(1)).deleteStock(argumentCaptor.capture());
+		Assert.assertEquals(argumentCaptor.getValue().longValue(), 1);
+	}
+
+	@Test
+	public void testGetStockItemsByServicePoint() throws Exception {
+		List<Stock> stocks = new ArrayList<>();
+		stocks.add(createStock());
+		when(stockService.getStocksByServicePointId(anyString())).thenReturn(stocks);
+		MvcResult result =
+				mockMvc.perform(MockMvcRequestBuilders.get(BASE_URL + "/servicePointId/{servicePointId}", "loc-1"))
+						.andExpect(MockMvcResultMatchers.status().isOk()).andReturn();
+
+		Mockito.verify(stockService)
+				.getStocksByServicePointId(ArgumentMatchers.any(String.class));
+		Mockito.verifyNoMoreInteractions(stockService);
+
+		String responseString = result.getResponse().getContentAsString();
+		if (responseString.isEmpty()) {
+			AssertionErrors.fail("Test case failed");
+		}
+		JsonNode actualObj = mapper.readTree(responseString);
+        assertEquals(1, actualObj.size());
+		Assert.assertEquals(12345l, actualObj.get(0).get("identifier").asLong());
+
+	}
+
+	@Test
+	public void testImportInventoryData() throws Exception {
+		Authentication authentication = mock(Authentication.class);
+		authentication.setAuthenticated(Boolean.TRUE);
+		SecurityContext securityContext = mock(SecurityContext.class);
+		SecurityContextHolder.setContext(securityContext);
+
+		when(SecurityContextHolder.getContext().getAuthentication()).thenReturn(getMockedAuthentication());
+		String path = "src/test/resources/sample/inventory.csv";
+		MockMultipartFile file = new MockMultipartFile("file", "sampleFile.txt", "text/csv",
+				Files.readAllBytes(Paths.get(path)));
+
+		CsvBulkImportDataSummary csvBulkImportDataSummary = new CsvBulkImportDataSummary();
+		csvBulkImportDataSummary.setNumberOfCsvRows(2);
+		csvBulkImportDataSummary.setNumberOfRowsProcessed(1);
+		List<FailedRecordSummary> failedRecordSummaryList = new ArrayList<>();
+		FailedRecordSummary failedRecordSummary = new FailedRecordSummary();
+		List<String> reasons = new ArrayList<>();
+		reasons.add("PO Number should be a whole number");
+		failedRecordSummary.setReasonOfFailure(reasons);
+		failedRecordSummary.setRowNumber(1);
+		failedRecordSummaryList.add(failedRecordSummary);
+		csvBulkImportDataSummary.setFailedRecordSummaryList(failedRecordSummaryList);
+
+		when(stockService.convertandPersistInventorydata(anyList(), anyString())).thenReturn(csvBulkImportDataSummary);
+
+		mockMvc.perform(
+				MockMvcRequestBuilders.multipart(BASE_URL + "/import/inventory")
+						.file(file)
+		)
+				.andExpect(status().isBadRequest())
+				.andReturn();
+
+	}
```

Comment:
Assert in the returned file, if its empty, also a test for upload that csv with errors